### ISSUE TITLE
All version page displays one license instead of all when navigating back

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-frontend#10597]
While testing for https://github.com/mozilla/addons-server/issues/17163 I've noticed that there's a display issue when a custom license is present on all versions page.
- click the custom license
- go back to all versions 

expected:
all versions for the add-on should be displayed

actual result:
only the last version is visible. The page needs a refresh

Notes:
I could not reproduce it with an add-on from production. I tried with https://addons.mozilla.org/en-US/firefox/addon/flagfox/versions/?utm_content=featured&utm_medium=referral&utm_source=addons.mozilla.org

I could reproduce it on AMO dev and AMO stage - with:
https://addons-dev.allizom.org/en-US/firefox/addon/firefox-to-all-applications/versions/
https://addons.allizom.org/en-US/firefox/addon/4manthease-mosel-codcanzee/versions/

![custom licenses](https://user-images.githubusercontent.com/33448286/118813169-8c75fa00-b8b7-11eb-94d2-f33deaf88872.gif)

**Comment by diox** [comment]
I don't think this blocks the push of https://github.com/mozilla/addons-server/issues/17163 but it has got to be related...

**Comment by bobsilverberg** [comment]
> I don't think this blocks the push of [mozilla/addons-server#17163](https://github.com/mozilla/addons-server/issues/17163) but it has got to be related...

I agree. I'll look into it. I know we do some fancy things with the versions reducer to deal with the fact that sometimes we have license info and sometimes we don't, so I assume it has something to do with that.

**Comment by bobsilverberg** [comment]
I can see what's happening here, and I'm not sure why https://github.com/mozilla/addons-server/issues/17163 would have had an effect, but I guess it's possible.

It looks like the call we make to `fetchVersion` for the licence page clears all of the versions for that add-on, which is why they are gone when we return to the all versions page. Perhaps the change that was made for https://github.com/mozilla/addons-server/issues/17163 somehow is preventing the all versions page from being able to re-fetch the list of versions?

I will dig into this some more next week.
